pkg/skaffold/runner/deployer.go

```diff
@@ -66,11 +66,42 @@ func (d *deployerCtx) JSONParseConfig() v1.JSONParseConfig {
 
 // GetDeployer creates a deployer from a given RunContext and deploy pipeline definitions.
 func GetDeployer(ctx context.Context, runCtx *runcontext.RunContext, labeller *label.DefaultLabeller) (deploy.Deployer, error) {
+	deployerCfg := runCtx.Deployers()
+
 	if runCtx.Opts.Apply {
+		helmNamespaceValue := ""
+		helmReleaseCounter := 0
+
+		for _, d := range deployerCfg {
+			if d.DockerDeploy != nil || d.KptDeploy != nil || d.KubectlDeploy != nil || d.KustomizeDeploy != nil {
+				helmReleaseCounter = -1
+				break
+			}
+
+			if d.HelmDeploy != nil {
+				for _, release := range d.HelmDeploy.Releases {
+					if release.Namespace != "" {
+						helmNamespaceValue = release.Namespace
+					}
+					helmReleaseCounter++
+				}
+			}
+		}
+
+		if helmNamespaceValue != "" && helmReleaseCounter != 1 {
+			return nil, errors.New("skaffold apply called with 'deploy.helm.releases.*.namespace' field set which is not supported.  Remove the 'deploy.helm.releases.*.namespace' field(s) and run skaffold apply again")
+		}
```

Comment:
It should be ok to call multiple releases with the same namespace right?

Reply:
Yes, I will add that functionality

Reply:
Added the suggested functionality